### PR TITLE
Provide skeletal API endpoints for functions on Byron-style wallets.  

### PR DESCRIPTION
**Pull request by jonathanknowles**
# Issue Number

#774 
#775 
#776 
#777 

# Overview

This PR provides skeletal API endpoints for a subset of the functions on Byron-style wallets.

The goal of this PR is to establish the general direction, including types and module organization, and not to provide a complete implementation. (This will be provided in further PRs.)

- [x] Separated the API into a core API (non-legacy) and a compatibility API (includes operations on Byron wallets).
- [x] Provided API types for a subset of functions on Byron-style wallets.
- [x] Provided compatibility API server implementation that returns HTTP 501 (Not Implemented) for each of the functions on Byron-style wallets.
- [x] Added appropriate definitions to the Swagger API specification.

### CODE REVIEW

**Comment by jonathanknowles** [comment]
bors try

File: lib/cli/src/Cardano/CLI.hs [hunk 1]
```diff
@@ -90,7 +90,7 @@ import Cardano.BM.Setup
 import Cardano.BM.Trace
     ( Trace, appendName, logAlert, logInfo )
 import Cardano.Wallet.Api
-    ( Api )
+    ( CoreApi )
```
Comment (jonathanknowles) [inline lib/cli/src/Cardano/CLI.hs:93]: We don't provide any support for operations on Byron wallets in the CLI, so there's no need to import the whole API: we can just import the core.

**Review by paweljakubas — APPROVED** [review]
LGTM!

**Pull request merged**